dev/attach_security_grp_to_port.template

```diff
@@ -0,0 +1,51 @@
+heat_template_version: 2015-04-30
+description: Test template to create port wit ip_address.
+parameters:
+  mac:
+    type: string
+    default: 00-00-00-00-BB-BB
+resources:
+  server1:
+    type: OS::Nova::Server
+    properties:
+      name: Server1
+      image: { get_param: image }
+      flavor: { get_param: flavor }
+      key_name: { get_param: key_name }
+      networks:
+        - port: { get_resource: server1_port }
+  net:
+    type: OS::Neutron::Net
+  subnet:
+    type: OS::Neutron::Subnet
+    properties:
+      network: { get_resource: net }
+      cidr: 11.11.11.0/24
+  server1_port:
+    type: OS::Neutron::Port
+    properties:
+      network: { get_resource: net }
+      mac_address: { get_param: mac }
+      fixed_ips:
+        - subnet: { get_resource: subnet }
+          ip_address: 11.11.11.11
+  security_grp:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      description: a security group to be added to server
+      name: security1
+      rules:
+        - remote_ip_prefix: 0.0.0.0/0
+          protocol: tcp
+          port_range_min: 22
+          port_range_max: 22
+        - remote_ip_prefix: 0.0.0.0/0
+          protocol: icmp
+  security_grp_attachment:
+    type: OS::Neutron::AttachSecurityGroupToPort
+    properties:
+      port: { get_resource: server1_port }
```

Comment:
This won't work and isn't the use case as I understand it. You should be looking for the public net port like:

```yaml
properties:
  port: { get_attr: [ server1, networks, public, port, id ] }
```